### PR TITLE
Returning microseconds on `NaiveDateTime.to_gregorian_seconds`

### PR DESCRIPTION
**Pull request by kelvinst**
Discussed in #10061

### CODE REVIEW

**Comment by josevalim** [comment]
Looks great! Can you also do the appropriate changes to `Time.to_seconds_after_midnight`? Thanks @kelvinst!

**Comment by josevalim** [comment]
Btw, if you want to add DateTime.to_gregorian_seconds/from_gregorian_seconds in another PR, it would be welcome!

**Comment by kelvinst** [comment]
> Looks great! Can you also do the appropriate changes to Time.to_seconds_after_midnight? 

Done!

**Comment by kelvinst** [comment]
> Btw, if you want to add DateTime.to_gregorian_seconds/from_gregorian_seconds in another PR, it would be welcome!

Doing :)

File: lib/elixir/lib/calendar/time.ex [hunk 1]
```diff
@@ -402,16 +402,16 @@ defmodule Time do
   ## Examples
```
Comment (fertapric) [inline lib/elixir/lib/calendar/time.ex:402]: We should also update the description of this function. For example:

```suggestion
  The returned value is a two-element tuple with the number of seconds and microseconds.

  ## Examples
```

File: lib/elixir/lib/calendar/time.ex [hunk 2]
```diff
@@ -402,16 +402,16 @@ defmodule Time do
...
       iex> Time.to_seconds_after_midnight(~T[23:30:15])
-      84615
+      {84615, 0}
       iex> Time.to_seconds_after_midnight(~N[2010-04-17 23:30:15.999])
-      84615
+      {84615, 999000}
 
   """
   @doc since: "1.11.0"
   @spec to_seconds_after_midnight(Calendar.time()) :: integer()
```
Comment (fertapric) [inline lib/elixir/lib/calendar/time.ex:411]:
```suggestion
  @spec to_seconds_after_midnight(Calendar.time()) :: {integer(), integer()}
```

Comment (kelvinst) [reply lib/elixir/lib/calendar/time.ex:411]: yep, remembered that when doing the other PR, will update

File: lib/elixir/lib/calendar/naive_datetime.ex [hunk 3]
```diff
@@ -814,18 +814,18 @@ defmodule NaiveDateTime do
...
-      1
+      {1, 0}
       iex> NaiveDateTime.to_gregorian_seconds(~N[2020-05-01 00:26:31.005])
-      63_755_511_991
+      {63_755_511_991, 5000}
 
   """
   @doc since: "1.11.0"
-  @spec to_gregorian_seconds(Calendar.naive_datetime()) :: integer()
+  @spec to_gregorian_seconds(Calendar.naive_datetime()) :: {integer(), integer()}
```
Comment (wojtekmach) [inline lib/elixir/lib/calendar/naive_datetime.ex:828]:
```suggestion
  @spec to_gregorian_seconds(Calendar.naive_datetime()) :: {integer(), non_neg_integer()}
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**